pkg/metrics/metrics.go

```diff
@@ -154,22 +156,65 @@ func (c *textFileCollector) Collect(ch chan<- prometheus.Metric) {
 
 // ProcessMetricsFile processes a metrics file that follows Prometheus text format: https://prometheus.io/docs/instrumenting/exposition_formats/,
 // returning its MetricFamily.
-func ProcessMetricsFile(path string) (map[string]*dto.MetricFamily, error) {
-	f, err := os.Open(path)
+func ProcessMetricsFile(directoryPath string) (map[string]*dto.MetricFamily, error) {
+	// Find latest file generation.
+	generationFilePath := directoryPath + generationFileName
```

Comment:
I would probably use the `filepath` library to construct the so that it's OS-independent

Reply:
Since this is already merged, will follow up on next PR and put PR # here.

Reply:
#336 contains this change.